src/lib/functions/netlify-function.ts

```diff
@@ -286,10 +286,11 @@ export default class NetlifyFunction {
    * Matches all routes agains the incoming request. If a match is found, then the matched route is returned.
    * @param {string} rawPath
    * @param {string} method
+   * @param {() => Promise<boolean>} hasStaticFile
```

Comment:
Since you're touching this function, can you add actual types to the parameters rather than expanding the JSDoc annotations?

Reply:
good idea 👍 will do that once https://github.com/netlify/cli/pull/6212 is merged, there'll probably be merge conflicts

Reply:
done! typescriptified a bunch of this file.